### PR TITLE
Persist mime type in SarifLogger. Remove stacks from configuration ex…

### PR DESCRIPTION
**Pull request by michaelcfanning**
…ception notifications. Reclassify 'could not parse target' as a configuration issue, not a tool error (since the most likely scenario is that an invalid target is being inspected, not that the tool parser is broken). @lgolding @rtaket

### CODE REVIEW

File: src/Sarif.UnitTests/JsonTests.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public JsonTests()
+        {
+            JsonConvert.DefaultSettings = () => new JsonSerializerSettings
+            {
+                ContractResolver = SarifContractResolver.Instance
+            };
+        }
+    }
+}
```
Comment (lgolding) [inline src/Sarif.UnitTests/JsonTests.cs]: Funny you should be doing this; I did the same thing in a branch recently. I suggest you move the `GetJson` method that's duplicated in a few test classes into this base class as well.

File: src/Sarif.UnitTests/Writers/ResultLogObjectWriterTests.cs [hunk 2]
```diff
@@ -1,84 +0,0 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/ResultLogObjectWriterTests.cs]: Why removed?

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs [hunk 3]
```diff
@@ -66,5 +68,42 @@ public void SarifLogger_RedactedCommandLine()
...
+        public void SarifLogger_WritesFileData()
+        {
+            var sb = new StringBuilder();
+            string file;
+
+            using (var tempFile = new TempFile(".cpp"))
+            using (var textWriter = new StringWriter(sb))            
+            {
+                File.WriteAllText(tempFile.Name, "#include \"windows.h\";");
+                file = tempFile.Name;
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: Reverse these two lines and then use `file` in the call to `WriteAllText`.

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs [hunk 4]
```diff
@@ -66,5 +68,42 @@ public void SarifLogger_RedactedCommandLine()
...
+
+                using (var sarifLogger = new SarifLogger(
+                    textWriter,
+                    analysisTargets: new string[] { tempFile.Name },
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: ... and use it here as well.

**Comment by lgolding** [comment]
:+1:

File: src/Sarif.UnitTests/Writers/ResultLogObjectWriterTests.cs (hunk 2 above)
Comment (michaelcfanning) [inline src/Sarif.UnitTests/Writers/ResultLogObjectWriterTests.cs]: matching product type was deleted. test file removed from project but lingered on disk

**Pull request merged**